net/record.go

```diff
@@ -38,22 +43,49 @@ func (r *recordCollector) Store(lid peer.ID, rec core.Record) {
 	seq.Store(rec)
 }
 
+func (r *recordCollector) UpdateHeadCounter(lid peer.ID, counter int64) {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	// we update the counter only if we have some records
+	val, found := r.counters[lid]
+	if !found {
+		r.counters[lid] = counter
+		// if a peer does not support the new logic we cannot rely on counter comparison
```

Comment:
uber nit: how about moving the comment into the block it refers to? This line is for the `if` below, but looks more closely to the code above it.